pkg/frontend/middleware/audit.go

```diff
@@ -0,0 +1,132 @@
+package middleware
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+	"github.com/Azure/ARO-RP/pkg/util/log/audit"
+)
+
+// Audit generates an audit log based on the request, caller, resource and
+// correlation data found in a HTTP request. It depends on the 'Log' middleware
+// to populate the request context with data that it needs.
+func Audit(env env.Core, entry *logrus.Entry) func(http.Handler) http.Handler {
```

Comment:
This file is quite intricate, and I wonder whether we can afford not to unit test it.  WDYT?